### PR TITLE
2469 document a dedicated page for experimental features

### LINKED ISSUE #2469: Document: a dedicated page for experimental features
be visible to both users and us.

### PR DESCRIPTION
**Pull request by ShanlanLi**
## Description

dedicated page for experimental features(added to the original product lifecycle page, which is applied to all versions.)
image
image

The public preview note looks likes this:
image

To do:

After the note content is fixed, replace all the existing beta notes with the new note.

## Related doc issue
resolves https://github.com/risingwavelabs/risingwave-docs/issues/2469

## Checklist

- [ ] I have checked the doc site preview, and the updated parts look good.
- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`emile-00`, `hengm3467`, & `WanYixian`).

### CODE REVIEW

**Comment by ShanlanLi** [comment]
@lmatz Please review the update. Feel free to tell me if there is anything to be improved.

By the way, how should we decide the start date of the public preview stage? The date that the code pr is merged?

File: src/pages/product-lifecycle.md [hunk 1]
```diff
@@ -5,16 +5,30 @@ description: RisingWave's product release lifecycle and how RisingWave defines e
...
+## List of features in the public preview stage
+
+As introduced above, when you see a "Public preview" note in the documentation, it indicates that a feature has not yet achieved 100% stability. We recommend evaluating the feature according to your specific use case. If you encounter any issues with the feature, please contact us via our [Slack channel](https://www.risingwave.com/slack). We also welcome any of your feedback to help us improve it.
+
+The following is a list of all features in the public preview phase:
+
+| Feature name                            | In public preview phase? | Start date | Start version | End date |
+|-----------------------------------------|-------------------------|------------|---------------|----------|
+| [Subscription](/docs/next/subscription/) | Yes                     | 2024-06-15 | 1.0.0         | \        |
+| Time travel queries                     | Yes                     | 2024-07-01 | 1.0.1         | \        |
```
Comment (lmatz) [inline src/pages/product-lifecycle.md]: how about also adding the link for the rest 3 features

**Review by lmatz — APPROVED** [review]
rest LGTM

File: src/pages/product-lifecycle.md [hunk 2]
```diff
@@ -5,16 +5,30 @@ description: RisingWave's product release lifecycle and how RisingWave defines e
...
 
-The beta stage is where we've incorporated feedback from the technical preview and are nearing the final product. It's more stable than the technical preview, but there may still be some bugs and issues.
+2. **Public preview**: The public preview stage is where we've incorporated feedback from the technical preview and are nearing the final product. It's more stable than the technical preview, but there may still be some bugs and issues. During this stage, corresponding guides are provided in the official documentation, and the feature is marked with a "Public preview" note. A complete list of all such features is offered below.
 
-## Generally available
+3. **General availability (GA)**: All other features are considered to be generally available. The general availability stage is where the feature is fully developed, tested, and ready for use in production environments. At this stage, the "Public preview" note is removed.
```
Comment (WanYixian) [inline src/pages/product-lifecycle.md]:
```suggestion
3. **General availability (GA)**: All other features are considered to be generally available. The general availability stage is where the feature is fully developed, tested, and ready for use in production environments.
```

File: src/pages/product-lifecycle.md [hunk 3]
```diff
@@ -5,16 +5,30 @@ description: RisingWave's product release lifecycle and how RisingWave defines e
...
+In RisingWave, the release of product typically follows three main stages:
 
-## Beta
+1. **Technical preview**: The technical preview stage is where we release a new feature for testing and feedback. It's not fully polished or finalized and is subject to change. We cannot guarantee its continued support in future releases, and it may be discontinued without notice. You may use this feature at your own risk.
 
-The beta stage is where we've incorporated feedback from the technical preview and are nearing the final product. It's more stable than the technical preview, but there may still be some bugs and issues.
+2. **Public preview**: The public preview stage is where we've incorporated feedback from the technical preview and are nearing the final product. It's more stable than the technical preview, but there may still be some bugs and issues. During this stage, corresponding guides are provided in the official documentation, and the feature is marked with a "Public preview" note. A complete list of all such features is offered below.
```
Comment (WanYixian) [inline src/pages/product-lifecycle.md]:
```suggestion
2. **Public preview**: The public preview stage is where we've incorporated feedback from the technical preview and are nearing the final product. It's more stable than the technical preview, but there may still be some bugs and issues. During this stage, the corresponding guides are available in the official documentation for your reference. For the overview of all such features, see the list below.
```

**Review by WanYixian — APPROVED** [review]
Rest LGTM

File: src/pages/product-lifecycle.md [hunk 4]
```diff
@@ -5,16 +5,30 @@ description: RisingWave's product release lifecycle and how RisingWave defines e
...
+This page lists RisingWave's product release lifecycle, how RisingWave defines each stage, and a complete list of all features in the public preview stage.
 
-## Technical preview
+## Product release lifecycle
 
-The technical preview stage is where we release a new feature for testing and feedback. It's not fully polished or finalized and is subject to change. We cannot guarantee its continued support in future releases, and it may be discontinued without notice. You may use this feature at your own risk.
+In RisingWave, the release of product typically follows three main stages:
 
-## Beta
+1. **Technical preview**: The technical preview stage is where we release a new feature for testing and feedback. It's not fully polished or finalized and is subject to change. We cannot guarantee its continued support in future releases, and it may be discontinued without notice. You may use this feature at your own risk.
```
Comment (ShanlanLi) [inline src/pages/product-lifecycle.md:14]: @lmatz I wonder what is the difference between our technical preview and the private preview? How can our users figure out such features? Should we do something to tag such features？

**Review by emile-00 — APPROVED** [review]

**Pull request merged**